Modules/IO/NIFTI/test/itkNiftiImageIOTest13.cxx

```diff
@@ -0,0 +1,52 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkNiftiImageIOTest.h"
+
+// Test to read small voxel NIFTI file which was triggering numerical instability
+// in IsAffine loading code
+int
+itkNiftiImageIOTest13(int argc, char * argv[])
+{
+  if (argc != 2)
+  {
+    return EXIT_FAILURE;
+  }
+
+  constexpr unsigned int Dimension = 3;
+
+  using PixelType = float;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  try
+  {
+    using ReaderType = itk::ImageFileReader<ImageType>;
+    ReaderType::Pointer reader = ReaderType::New();
+    reader->SetFileName(argv[1]);
+    reader->Update();
+    ImageType::Pointer image = reader->GetOutput();
+  }
+  catch (itk::ExceptionObject & err)
```

Comment:
Please catch by "const reference", if possible.

Reply:
> @hjmjohnson I replaced the use of itkNiftiImageIOTest2 with my own test.

Would it not be possible to re-use itkNiftiImageIOTest2 in this case (instead of adding an extra cxx file)?

Reply:
> That's far beyond my C++ experience to know what you're asking.

@gdevenyi  When you're programming C++, you should really know about the difference between "const reference" and non-const reference.

By default, it is recommended in C++ to catch exceptions by const reference:

    catch (const itk::ExceptionObject & err)

As a rule of thumb, only use a non-const reference, when you intend to modify the exception object that you have "caught". 

HTH, Niels

Reply:
Done.